src/toil/test/__init__.py

```diff
@@ -394,23 +394,31 @@ def insertMethodToClass():
         Generates and inserts test methods.
         """
         def fx(self, prms=prms):
-            return generalMethod(self, **prms)
+            if prms is not None:
+                return generalMethod(self, **prms)
+            else:
+                return generalMethod(self)
         setattr(targetClass, 'test_%s%s' % (generalMethod.__name__, prmNames), fx)
 
-    # create first left dict
-    left = {}
-    prmName, vals = pop(kwargs)
-    for valName, val in vals.items():
-        pvName = '__%s_%s' % (prmName, valName.title())
-        if methodNamePartRegex.match(pvName) is None:
-            raise RuntimeError("The name '%s' cannot be used in a method name" % pvName)
-        left[pvName] = {prmName: val}
-
-    # get cartesian product
-    while len(kwargs) > 0:
-        permuteIntoLeft(left, *pop(kwargs))
-
-    # set class attributes
-    targetClass = targetClass or generalMethod.im_class
-    for prmNames, prms in left.items():
+    if len(kwargs) > 0:
```

Comment:
The import only tests (which currently only tests for http urls) does not require any parameter permutations because the inheriting jobStore class will just import from some http url. The change in make_tests() allows a test method to be generated without any general method parametrization  
```
 @classmethod
        def makeImportOnlyTests(cls):
            def importHttpFile(self):
                # prepare random file for import
                self.master.partSize = cls.mpTestPartSize
                srcUrl = 'https://raw.githubusercontent.com/BD2KGenomics/toil/master/Makefile'
                srcHash = hashlib.md5(urllib2.urlopen(srcUrl).read()).hexdigest()
                # test import
                jobStoreFileID = self.master.importFile(srcUrl)
                self.assertEqual(self._hashJobStoreFileID(jobStoreFileID), srcHash)

            make_tests(importHttpFile, targetClass=cls)
```